src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c

```diff
@@ -1015,8 +1015,80 @@ zocl_load_aie_only_pdi(struct drm_zocl_dev *zdev, struct axlf *axlf,
 int
 zocl_pl_only_reset(struct drm_zocl_dev *zdev, const char *buf, size_t count)
 {
-	/* TODO */
-	DRM_ERROR("%s PL Reset support doesn't added yet", __func__);
+	const struct list_head *ptr = NULL;
+	struct drm_zocl_slot *slot = zdev->pr_slot[ZOCL_DEFAULT_XCLBIN_SLOT];
+	struct kds_sched *kds = NULL;
+	struct kds_client *client = NULL;
+	struct kds_client_ctx *curr = NULL;
+	struct kds_client_ctx *tmp = NULL;
+	void __iomem *map = NULL;
+	int ret = 0;
+
+	if (!slot)
+	       return count;
+
+	if (slot->xclbin_type == ZOCL_XCLBIN_TYPE_FULL)
+		return count;
+
+	if (strcmp(zdev->zdev_data_info->fpga_driver_name, "versal_fpga"))
+		return count;
+
+	mutex_lock(&slot->slot_xclbin_lock);
+	kds = &zdev->kds;
+
+	/* Find out number of active client and free all the context
+	 * associated with it
+	 */
+	list_for_each(ptr, &kds->clients) {
```

Comment:
When iterate kds->clients list, think about if we need to hold kds->lock. Can slot_xclbin_lock protect kds->clients?
Similar locking question for client->ctx_list.

Reply:
The implementation has been changed. Please re-review once again.